Design/Algorithm_ease_of_use.md

```diff
@@ -0,0 +1,88 @@
+Motivation
+==========
+
+Currently code (especially workflow algorithms) will be full of things
+like
+
+```C
+double tolerance = getProperty("CompressTolerance");
+m_outputEW = boost::dynamic_pointer_cast<EventWorkspace>(m_outputW);
+if ((m_outputEW) && (tolerance > 0.)) {
+  g_log.information() << "running CompressEvents(Tolerance=" << tolerance
+                      << ")\n";
+  API::IAlgorithm_sptr compressAlg = createChildAlgorithm("CompressEvents");
+  compressAlg->setProperty("InputWorkspace", m_outputEW);
```

Comment:
Sure, and I know that you deliberately chose a less than perfect implementation to illustrate the point. As I was looking at it I was wondering if it actually makes sense for us to think of Algorithms as classes. They're more like functions, and I was wondering if taking a functional approach to `exec` and below would be beneficial.